### PR TITLE
Add bibox fetch deposits/withdrawals

### PR DESCRIPTION
**Pull request by lottaeouss**

### CODE REVIEW

File: js/bibox.js [hunk 1]
```diff
@@ -422,6 +424,127 @@ module.exports = class bibox extends Exchange {
...
+        //      'confirmCount': '16',
+        //      'createdAt': 1553123867000,
+        //      'status': 2}]                                                             }
+        //
+        // fetchWithdrawals
+        //
+        //       {      amount:  14,
+        //             address: "0x0123456789abcdef...",
+        //         successTime:  1514489710000,
+        //          addressTag: "",
```
Comment (kroitor) [inline js/bibox.js]: We will also need the tag here.

File: js/bibox.js [hunk 2]
```diff
@@ -422,6 +424,127 @@ module.exports = class bibox extends Exchange {
...
+        let status = this.parseTransactionStatusByType (this.safeString (transaction, 'status'), type);
+        let amount = this.safeFloat (transaction, 'amount');
+        return {
+            'info': transaction,
+            'id': id,
+            'txid': undefined,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'address': address,
+            'tag': undefined,
```
Comment (kroitor) [inline js/bibox.js]: The `tag` goes here.

File: js/bibox.js [hunk 3]
```diff
@@ -422,6 +424,124 @@ module.exports = class bibox extends Exchange {
...
+        let symbol = undefined;
+        if (code !== undefined) {
+            const currency = this.currency (code);
+            symbol = currency['id'];
+        }
+        let size = (limit) ? limit : 100;
+        let response = await this.privatePostTransfer ({
+            'cmd': 'transfer/transferInList',
+            'body': this.extend ({
+                'search': symbol,
```
Comment (kroitor) [inline js/bibox.js]: If this thing is required, then the `code` cannot be undefined. In that case we should throw an `ArgumentsRequired` exception, for example: https://github.com/ccxt/ccxt/blob/master/js/coinex.js#L773

File: js/bibox.js [hunk 4]
```diff
@@ -422,6 +424,124 @@ module.exports = class bibox extends Exchange {
...
     }
 
+    async fetchDeposits (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let symbol = undefined;
+        if (code !== undefined) {
+            const currency = this.currency (code);
+            symbol = currency['id'];
+        }
+        let size = (limit) ? limit : 100;
```
Comment (kroitor) [inline js/bibox.js]: If `size` is not required, we should not send it. I mean, if it is not required, we should only send it if it's undefined.

File: js/bibox.js [hunk 5]
```diff
@@ -422,6 +424,124 @@ module.exports = class bibox extends Exchange {
...
+            code = currency['code'];
+        }
+        let timestamp = this.safeString (transaction, 'createdAt');
+        let type = this.safeString (transaction, 'type');
+        let status = this.parseTransactionStatusByType (this.safeString (transaction, 'status'), type);
+        let amount = this.safeFloat (transaction, 'amount');
+        return {
+            'info': transaction,
+            'id': id,
+            'txid': undefined,
```
Comment (kroitor) [inline js/bibox.js]: We need txid here.

File: js/bibox.js (hunk 3 above)
Comment (lottaeouss) [reply js/bibox.js]: code can be left empty have changed the request body structure

File: js/bibox.js (hunk 4 above)
Comment (lottaeouss) [reply js/bibox.js]: for bibox, it seems like the size and pages fields are required, e.g. if they are not sent with the request, this error is thrown: 
ccxt.base.errors.ExchangeError: bibox {"error":{"code":"3000","msg":"请求参数错误"},"cmd":"transfer/transferInList"}

File: js/bibox.js (hunk 5 above)
Comment (lottaeouss) [reply js/bibox.js]: txid is not provided by bibox, so we should remove it?

        //       {'id': 521844,
        //       'coin_symbol': 'ETH',
        //       'to_address': '0xfd4e....',
        //       'addr_remark': '',
        //       'amount': '0.39452750',
        //       'fee': '0.00600000',
        //       'createdAt': 1553226906000,
        //       'memo': '',
        //       'status': 3}
        //

Comment (kroitor) [reply js/bibox.js]: Is that a withdrawal structure or a deposit structure? Can you paste a withdrawal structure here?

> so we should remove it?

Nope, we should keep it undefined if there's no `txid` in withdrawals.

Comment (lottaeouss) [reply js/bibox.js]: ok I have added it back thanks

Comment (lottaeouss) [reply js/bibox.js]: > Is that a withdrawal structure or a deposit structure? Can you paste a withdrawal structure here?
> 
> > so we should remove it?
> 
> Nope, we should keep it undefined if there's no `txid` in withdrawals.

structures as below if you haven't seen already.
        //
        // fetchDeposits
        //      {'id': 1023291,
        //      'coin_symbol': 'ETH',
        //      'to_address': '0x7263....',
        //      'amount': '0.49170000',
        //      'confirmCount': '16',
        //      'createdAt': 1553123867000,
        //      'status': 2}                                                             }
        //
        // fetchWithdrawals
        //
        //       {'id': 521844,
        //       'coin_symbol': 'ETH',
        //       'to_address': '0xfd4e....',
        //       'addr_remark': '',
        //       'amount': '0.39452750',
        //       'fee': '0.00600000',
        //       'createdAt': 1553226906000,
        //       'memo': '',
        //       'status': 3}
        //

**Comment by lottaeouss** [comment]
Any more changes needed to be made?

**Comment by kroitor** [comment]
@lottaeouss nope, i'll merge this shortly. Thx for your effort and pardon for the delay on my side.

**Pull request merged**